crates/walrus-service/src/node/storage/shard.rs

```diff
@@ -1066,6 +1080,11 @@ impl ShardStorage {
 
                 batch.write()?;
 
+                // Clear the recovery deferral now that we have definitively synchronized the blob.
+                for blob_id in cleared_blob_ids {
+                    node.clear_recovery_deferral(&blob_id).await;
```

Comment:
Important: I don't think we should clear deferral here. Blob sync is for all shards, but here this only indicate that one shard is synced. There can be other shards that are missing.